### PR TITLE
refactor: preliminary cleanup of Deno.runTests()

### PR DESCRIPTION
**Pull request by bartlomieju**
see commit message

### CODE REVIEW

**Comment by bartlomieju** [comment]
Ref #4166

File: cli/js/testing.ts [hunk 1]
```diff
@@ -185,23 +200,27 @@ export async function runTests({
...
-    setTimeout((): void => {
-      originalConsole.error(`There were ${stats.failed} test failures.`);
-      testCases
-        .filter(testCase => !!testCase.error)
-        .forEach(testCase => {
-          originalConsole.error(`${RED_BG_FAIL} ${red(testCase.name)}`);
-          originalConsole.error(testCase.error);
-        });
+  // TODO(bartlomieju): is it really needed? Shouldn't that be handled
+  // per test case?
```
Comment (ry) [inline cli/js/testing.ts]: What does "it" and "that" refer to here?

Comment (bartlomieju) [reply cli/js/testing.ts]: The comment right below; must have inserted empty line by mistake

File: cli/js/testing.ts [hunk 2]
```diff
@@ -149,30 +159,35 @@ export async function runTests({
   const RED_BG_FAIL = bgRed(" FAIL ");
 
   originalConsole.log(`running ${testsToRun.length} tests`);
-  const suiteStart = performance.now();
+  // NOTE(bartlomieju): originally we used `performance.now()` here,
+  // but there's no guarantee that user runs tests with `--allow-hr`
+  // flag - so `new Date()` provides actually more precision
+  // (1ms instead of 2ms).
```
Comment (ry) [inline cli/js/testing.ts]: Comment seems unnecessary...

**Pull request merged**